sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/ClientLogger.java

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * This is a fluent logger helper class that wraps a pluggable {@link Logger}.
+ *
+ * <p>This logger logs formattable messages that use {@code {}} as the placeholder. When a {@link Throwable throwable}
+ * is the last argument of the format varargs and the logger is enabled for
+ * {@link ClientLogger#verbose(String, Object...) verbose}, the stack trace for the throwable is logged.</p>
+ *
+ * <p><strong>Log level hierarchy</strong></p>
+ * <ol>
+ * <li>{@link ClientLogger#error(String, Object...) Error}</li>
+ * <li>{@link ClientLogger#warning(String, Object...) Warning}</li>
+ * <li>{@link ClientLogger#info(String, Object...) Info}</li>
+ * <li>{@link ClientLogger#verbose(String, Object...) Verbose}</li>
+ * </ol>
+ *
+ */
+public class ClientLogger {
+    private final Logger logger;
+
+    /*
+     * Indicates that log level is at verbose level.
+     */
+    private static final int VERBOSE_LEVEL = 1;
```

Comment:
In Android, people tend to avoid using enums in favor of static int values with conventional names. My understanding is that the reasons have primarily to do with memory use (enums are objects), performance (enums used to be really slow on the older Dalvik runtime), and ease of use in IPC (passing value types when calling other services via Binder is much easier because you can just add them directly to the parcel - sending objects in a parcel is much more complicated and inefficient).

However, we should be using @IntDef / @StringDef annotations to provide enum-like validation checking at compile time though. See https://developer.android.com/reference/androidx/annotation/IntDef - also highly recommend just browsing through the list of annotations provided by the AndroidX library, many of them are ones that we should be using to help our customers better understand the contract of our APIs.